### PR TITLE
Implement IsInf

### PR DESCRIPTION
**Pull request by yuslepukhin**

### CODE REVIEW

File: onnxruntime/core/providers/cpu/tensor/isinf.cc [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  } else {
+    // all false
+    memset(output_data, false, total_items);
+  }
+}
+}  // namespace isinf_internal
+
+Status IsInf::Compute(OpKernelContext* context) const {
+  const Tensor* X_ptr = context->Input<Tensor>(0);
+  auto& X = *X_ptr;
```
Comment (hariharans29) [inline onnxruntime/core/providers/cpu/tensor/isinf.cc]: it: `const` for lines72 & 73 maybe ?

File: onnxruntime/core/providers/cpu/tensor/isinf.cc [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  auto& Y = *context->Output(0, shape);
+
+  using namespace isinf_internal;
+
+  if (X.DataType() == DataTypeImpl::GetType<float>()) {
+    ComputeImpl<float>(X, Y, detect_positive_ != 0, detect_negative_ != 0);
+  } else if (X.DataType() == DataTypeImpl::GetType<double>()) {
+    ComputeImpl<double>(X, Y, detect_positive_ != 0, detect_negative_ != 0);
+  } else {
+    return Status(common::ONNXRUNTIME, common::INVALID_ARGUMENT, "Unsupported input data type");
```
Comment (hariharans29) [inline onnxruntime/core/providers/cpu/tensor/isinf.cc]: verbosify the error message a bit like - ("supports float and double but instead got ", dtype)

**Comment by snnn** [comment]
Please also update https://github.com/Microsoft/onnxruntime/blob/master/onnxruntime/test/python/onnx_backend_test_series.py, revert the changes made in d78c340eac0dc251c1e96378fc1dab6e6cf8fdcc

File: onnxruntime/core/providers/cpu/tensor/isinf.cc [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+  auto output_data = Y.template MutableData<bool>();
+
+  if (detect_positive && detect_negative) {
+    EigenMap<bool>(Y) = EigenMap<T>(X).array().isInf();
+  } else if (detect_positive) {
+    auto input_data = X.template Data<T>();
+    auto end_data = input_data + total_items;
+    std::transform(
+        input_data, end_data, output_data, [](auto v) {
+          return (std::isinf(v) && !(v < std::numeric_limits<T>::lowest()));
```
Comment (wschin) [inline onnxruntime/core/providers/cpu/tensor/isinf.cc]: This probably can be simplified to `v == -std::numeric_limits<double>::infinity()`.

File: onnxruntime/core/providers/cpu/tensor/isinf.cc [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+        });
+
+  } else if (detect_negative) {
+    auto input_data = X.template Data<T>();
+    auto end_data = input_data + total_items;
+    std::transform(
+        input_data, end_data, output_data, [](auto v) {
+          return (std::isinf(v) && (v < std::numeric_limits<T>::lowest()));
```
Comment (wschin) [inline onnxruntime/core/providers/cpu/tensor/isinf.cc]: There is a `std::numeric_limits<double>::infinity()`. Maybe we can consider to merge the 3 operations here into a `==`.

File: onnxruntime/core/providers/cpu/tensor/isinf.cc (hunk 2 above)
Comment (wschin) [reply onnxruntime/core/providers/cpu/tensor/isinf.cc]: If possible, please also print out the name of the wrong tensor.

File: onnxruntime/core/providers/cpu/tensor/isinf.cc (hunk 3 above)
Comment (yuslepukhin) [reply onnxruntime/core/providers/cpu/tensor/isinf.cc]: No, it can not.

File: onnxruntime/core/providers/cpu/tensor/isinf.cc (hunk 2 above)
Comment (yuslepukhin) [reply onnxruntime/core/providers/cpu/tensor/isinf.cc]: Do you want to print a pointer?

---
In reply to: [277120346](https://github.com/Microsoft/onnxruntime/pull/871#discussion_r277120346) [](ancestors = 277120346)

Comment (yuslepukhin) [reply onnxruntime/core/providers/cpu/tensor/isinf.cc]: There is only one input, it is not?

---
In reply to: [277352672](https://github.com/Microsoft/onnxruntime/pull/871#discussion_r277352672) [](ancestors = 277352672)

File: onnxruntime/core/providers/cpu/tensor/isinf.cc (hunk 3 above)
Comment (yuslepukhin) [reply onnxruntime/core/providers/cpu/tensor/isinf.cc]: Do you mean v == std::numeric_limits<double>::infinity()?

---
In reply to: [277349556](https://github.com/Microsoft/onnxruntime/pull/871#discussion_r277349556) [](ancestors = 277349556)

File: onnxruntime/core/providers/cpu/tensor/isinf.cc (hunk 4 above)
Comment (yuslepukhin) [reply onnxruntime/core/providers/cpu/tensor/isinf.cc]: Two operations?

---
In reply to: [277349892](https://github.com/Microsoft/onnxruntime/pull/871#discussion_r277349892) [](ancestors = 277349892)

File: onnxruntime/core/providers/cpu/tensor/isinf.cc (hunk 2 above)
Comment (hariharans29) [reply onnxruntime/core/providers/cpu/tensor/isinf.cc]: refer to this- https://github.com/Microsoft/onnxruntime/blob/0bf12e9dbf7d5ba434dcaa8254a9495b9d3d7bd7/onnxruntime/core/providers/cpu/tensor/slice.cc#L272

**Pull request merged**